### PR TITLE
Add support to nodejs10.x and fix a bug when inferring nodejs lambda

### PR DESCRIPTION
**Pull request by zhangzhx**
handler source root.

When the target potential lambda handler is from the dependency lib, the content root cannot be inferred. It should return null for it's source root instead of throwing exception.

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamCommon.kt [hunk 1]
```diff
@@ -32,7 +32,7 @@ class SamCommon {
         const val SAM_INVALID_OPTION_SUBSTRING = "no such option"
 
         // Inclusive
-        val expectedSamMinVersion = SemVer("0.14.1", 0, 14, 1)
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamCommon.kt]: This will need a changelog entry of just like we did when we raised from 0.7,

I'd make a dedicated commit for this change

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/wizard/SamInitProjects.kt [hunk 2]
```diff
@@ -91,8 +91,7 @@ class SamDynamoDBCookieCutter : SamPythonProjectTemplate() {
 }
 
 abstract class SamNodeJsProjectTemplate : SamProjectTemplate() {
-    // TODO support NodeJs 10.10 at some point
-    override fun supportedRuntimes(): Set<Runtime> = setOf(Runtime.NODEJS8_10)
+    override fun supportedRuntimes(): Set<Runtime> = setOf(Runtime.NODEJS8_10, Runtime.NODEJS10_X)
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/ui/wizard/SamInitProjects.kt]: Hmmm should we just say NodeJsRuntimeGroup.runtimes()?

Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/wizard/SamInitProjects.kt]: No, I think we should be explicit here since we're also stating node8 and 10.x elsewhere

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/wizard/SamInitProjects.kt]: k

Comment (zhangzhx) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/wizard/SamInitProjects.kt]: For a long term, we need to load these configs dynamically. The template author should provide what runtimes it supports.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamCommon.kt (hunk 1 above)
Comment (zhangzhx) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamCommon.kt]: The whole nodejs feature branch will be merged into master in one commit. Is it worth having a separate commit for the min version bump?
